plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -195,14 +361,55 @@ public void outputFailedPlugins() {
      */
     public void downloadPlugins(List<Plugin> plugins) {
         for (Plugin plugin : plugins) {
-            boolean successfulDownload = downloadPlugin(plugin);
+            boolean successfulDownload = downloadPlugin(plugin, null);
             if (!successfulDownload) {
                 System.out.println("Unable to download " + plugin.getName() + ". Skipping...");
                 failedPlugins.add(plugin);
             }
         }
     }
 
+    /**
+     *
+     * @param requestedPlugins
+     * @return
+     */
+    public Map<String, Plugin> findPluginsAndDependencies(List<Plugin> requestedPlugins) {
+        Map<String, Plugin> allPluginDependencies = new HashMap<>();
+
+        for (Plugin requestedPlugin : requestedPlugins) {
+            //for each requested plugin, find all the dependent plugins that will be downloaded (including requested plugin)
```

Comment:
+1 just as an aside, feel free to markup what the methods are doing.  It'll be very helpful for collaborators in the repo if you add notes in the code.  Think about what could help someone else who's coming into the codebase with fresh eyes; what would help them figure out what's going on quickly.